libs/minecraft/PosUtils.arucas

```diff
@@ -136,4 +175,249 @@ enum Direction {
+        dz = otherPos.getZ() - this.z;
+        return (dx ^2 + dy ^2 + dz ^2) ^ 0.5;
+    }
+    fun asCenter(){
+        return new PosWithFun(this.x + 0.5, this.y + 0.5, this.z + 0.5);
+    } // Explicit same type casting
+
+    fun getSidePos(direction : Direction){ //for Wall Mounted Blocks vanilla protocol
+        //we expect Direction object here
+        return this.asCenter().offset(direction, 0.5);
+    } // Explicit same type casting
+    fun mult(integer : Number){
+        return new PosWithFun(this.x * integer, this.y * integer, this.z * integer);
+    }
+    fun asPos(){
+        return new Pos(this.x, this.y, this.z);
+    } //explicit casting
+    fun asJavaPos(){
+        return Java.constructClass('net.minecraft.util.math.BlockPos',this.x, this.y, this.z);
+    }//conversion
+    fun formatted(){
+        return 'x '+ this.x + ' y' + this.y + ' z' + this.z;
+    }
+    fun copy(){
+        return new PosWithFun(this.x, this.y, this.z);
+    }
+    fun offset(direction : Direction){
+        //south +z north -z west -x east +x
+        return this.add(direction.asVector());
+    }
+    fun offset(direction: Direction, amount : Number){
+        return this.internalAdd(direction, amount);
+    }
+    fun multiply(amount : Number){
+        //compat
+        return this.mult(amount);
+    }
+    fun up(){
+        return this.add(Direction.UP);
+    }
+    fun down(){
+        return this.add(Direction.DOWN);
+    }
+    fun up(int : Number){
+        return this.offset(Direction.UP, int);
+    }
+    fun down(int : Number){
+        return this.offset(Direction.DOWN, int);
+    }
+    fun east(){
+        return this.add(Direction.EAST);
+    }
+    fun west(){
+        return this.add(Direction.WEST);
+    }
+    fun east(int : Number){
+        return this.offset(Direction.EAST, int);
+    }
+    fun west(int : Number){
+        return this.offset(Direction.WEST, int);
+    }
+    fun north(){
+        return this.add(Direction.NORTH);
+    }
+    fun south(){
+        return this.add(Direction.SOUTH);
+    }
+    fun north(int : Number){
+        return this.offset(Direction.NORTH, int);
+    }
+    fun isNear(entity : Entity, distance : Number){
+        pos = entity.getPos();
+        return this.distanceTo(pos) < distance;
+    }
+
+    fun south(int : Number){
+        return this.offset(Direction.SOUTH, int);
+    }
+    fun getMinPos(otherPos : Pos){ //Compare two positions to get Minimum position (Box)
+        return new PosWithFun(Math.min(this.x, otherPos.x),Math.min(this.y, otherPos.y),Math.min(this.z, otherPos.z) );
+    }
+    fun getMaxPos(otherPos : Pos){
+        return new PosWithFun(Math.max(this.x, otherPos.x),Math.max(this.y, otherPos.y),Math.max(this.z, otherPos.z) );
+    }
+    fun equals(otherPos : Pos){
+        return this.x == otherPos.x && this.y == otherPos.y && this.z == otherPos.z;
+    }
+}
+class PosWithDirection : Pos{
+    // Utility class that holds Position and Direction information
```

Comment:
I don't think this is a good implementation of this.
You should make a wrapper class that stores a Position and a Direction, this is just kinda weird.